sql/core/src/test/resources/sql-tests/inputs/cte.sql

```diff
@@ -28,6 +28,20 @@ FROM   CTE1 t1
 WITH t(x) AS (SELECT 1)
 SELECT * FROM t WHERE x = 1;
 
+-- CTE with multiple column aliases
+WITH t(x, y) AS (SELECT 1, 2)
+SELECT * FROM t WHERE x = 1 AND y = 2;
+
+-- CTE with empty column alias list is not allowed
+WITH t() AS (SELECT 1)
+SELECT * FROM t;
+
+-- CTE with duplicate name is not allowed
```

Comment:
According to @gatorsmile 's comment, could you add a `t(x, x)` test case?
> Duplicate names within a single CTE definition are not allowed.

Reply:
We need to capture the duplicate names earlier and issue a better error message

Reply:
Hmm, I added the `t(x, x)` case, but it is allowed, and I think it is right to be allowed. 
I thought @gatorsmile referred to the
```
WITH
  t(x) AS (SELECT 1),
  t(x) AS (SELECT 2)
SELECT * FROM t;
```
case, which is not.

@gatorsmile we capture the duplicate CTE name during parsing, I fixed the error message though.

If I got something wrong please let me know.

Reply:
@gatorsmile, @dongjoon-hyun do you think the PR is ok now?